### ISSUE TITLE
FE: Map Display API 06/12-06/14

### ISSUE THREAD

**Issue by stripes416** [Plantalytics/plantalytics-frontend#13]
Set up specified map [vineyard = to add] for the demo.
06/12/16 - 06/14/16

Retrieve map location via faked or real AJAX call (depending on back-end implementation status). Fetch map from Google APIs and draw the given boundaries from the JSON received.

Drawing the boundaries may require using Leaflet so that should be investigated first. Only bother if that much can be set up without a heatmap, otherwise it can be part of the heatmap ticket. #14 
- [ ] Map location
- [ ] Add visual boundaries (if possible), either way the JSON should have boundaries indicated.

**Comment by kbrimm** [comment]
1. It is really easy to draw a property outline with a few Lat/Lon coordinates, and it doesn't dramatically affect load times.
2. The IDW plugin does not support masking/boundary limits at this time.
3. The IDW plugin IS open source, and boundaries are on its list of "to-do" items, so we could contribute to that functionality if we really wanted to.
4. That sounds like a lot of work, and possibly is out-of-scope for this iteration.

**Comment by logicplace** [comment]
Looking over Leaflet there's a couple things I want to bring up.

There's the fact that IDW isn't native in Leaflet to begin with so there's not necessarily a reliance on it there. Also, Leaflet doesn't seem to support satellite images directly and has to piggyback Google Maps anyway. Google Maps API has as part of their usage agreement that you do not use a third-party API or alternate means of accessing their images so we would have to use the GMaps API in some fashion anyway (note: I did find a means to hook the two together for this, but I also found a method that looked like it was in violation of the agreement).

Additionally, there is an IDW option for GMaps directly here: https://darrenwiens.wordpress.com/tag/inverse-distance-weighting/
which is drawn on a canvas that we could use CSS clipping to shape to our boundaries. I'm guessing that would be easier to do than implementing boundaries in Leaflet.idw but I couldn't say for sure.

I'm not really arguing for this method either just felt like I should bring it up.

**Comment by SoySauceLabs** [comment]
Excellent find, Sapphire! If we can go all Google, while making it easier on ourselves and/or be more flexible annnddd not break any agreements, than You've got my vote!! Oh, and interesting note Katy, the exponent on the IDW code from Sapphire's linke is set to 3. ;)

**Comment by kbrimm** [comment]
Thanks for doing the legwork on this! The note about the API usage is pertinent - I'd prefer to stay on Google's good side for the day they inevitably become the overlords in our dystopian future.

**Comment by kbrimm** [comment]
Reached out to both Google Maps and MapQuest in regards to API usage plan clarification.

**Comment by kbrimm** [comment]
After some experimentation and research, I think Leaflet is the way to go - it's very user-friendly and has a lot of third party integrations that support the utility we need. It also allows for the use of non-Google maps which are generally less expensive. MapQuest, for example, is definitively free-to-use up to 15,000 calls/month.

Going forward, implementation will be Leaflet/MapQuest.

**Comment by kbrimm** [comment]
Where are we on this task? Is it ready for QA, or blocked by something?

**Comment by ghost** [comment]
There was a couple lines of code that didn't make the conversion to leaflet. Since I didn't know their intended purpose, I commented them out (though I do know they were something for mobile interfaces). Sapphire said she will take care of those. Otherwise, the integration is finished and should be ready for QA.

**Comment by kbrimm** [comment]
Great! Since Sapphire's likely to be tied up for a bit, would you be willing to submit a pull request on this as-is?

**Comment by ghost** [comment]
Sure thing!
